basket/news/views.py

```diff
@@ -939,3 +955,36 @@ def update_user_task(request, api_call_type, data=None, optin=False, sync=False)
         return HttpResponseJSON({
             'status': 'ok',
         })
+
+
+@require_POST
+@csrf_exempt
+def subhub_post(request):
+    if not has_valid_api_key(request):
+        return HttpResponseJSON({
+            'status': 'error',
+            'desc': 'requires a valid API-key',
+            'code': errors.BASKET_AUTH_ERROR,
+        }, 401)
+
+    try:
+        data = json.loads(request.body)
+    except ValueError as _:
+        statsd.incr('subhub_post.message.json_error')
+        sentry_client.captureException(data={'extra': {'request.body': request.body}})
+
+        return HttpResponseJSON({
+            'status': 'error',
+            'desc': 'JSON error',
+            'code': errors.BASKET_USAGE_ERROR,
+        }, 400)  # TODO: find the best status code for this
+    else:
+        etype = data['event_type']
+        processor = SUBHUB_EVENT_TYPES.get(etype)
+
+        if processor:
+            processor.delay(data)
+            return HttpResponseJSON({'status': 'ok'})
+        else:
+            # TODO: set a better HTTP response code? 409?
+            return HttpResponseJSON({'status': 'unknown event type'})
```

Comment:
At least return a 400 and a status of `error` and this as `desc`. This is good though 👍